From d48fac7458e7a029c325e76ced028fd0b0afb44c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Mon, 28 Aug 2017 02:14:25 +0200
Subject: [PATCH] libxl: do not force qdisk backend for cdrom
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

There is no qemu in dom0 in Qubes, every block device use vbd backend
type and only then is appropriately handled by stubdomain.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 tools/libxl/libxl.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/tools/libxl/libxl.c b/tools/libxl/libxl.c
index 814aa02..5cca657 100644
--- a/tools/libxl/libxl.c
+++ b/tools/libxl/libxl.c
@@ -2095,17 +2095,6 @@ int libxl__device_disk_setdefault(libxl__gc *gc, libxl_device_disk *disk,
     rc = libxl__resolve_domid(gc, disk->backend_domname, &disk->backend_domid);
     if (rc < 0) return rc;
 
-    /* Force Qdisk backend for CDROM devices of guests with a device model. */
-    if (disk->is_cdrom != 0 &&
-        libxl__domain_type(gc, domid) == LIBXL_DOMAIN_TYPE_HVM) {
-        if (!(disk->backend == LIBXL_DISK_BACKEND_QDISK ||
-              disk->backend == LIBXL_DISK_BACKEND_UNKNOWN)) {
-            LOG(ERROR, "Backend for CD devices on HVM guests must be Qdisk");
-            return ERROR_FAIL;
-        }
-        disk->backend = LIBXL_DISK_BACKEND_QDISK;
-    }
-
     rc = libxl__device_disk_set_backend(gc, disk);
     return rc;
 }
-- 
2.9.5

